### PR TITLE
Updating view functions to query DB - WIP

### PR DESCRIPTION
**Pull request by amymhaddad**
@paulghaddad I'm working on running a book query to get the books from my DB, and have a a few questions:

1. In `app.py:`

 In the SQLAlchemy-Flask tutorial, it says to import db_session from my database file). This is what I have in app.py
`from upcase_booklist_app.database import db_session`

But I get an error here, and I'm not sure why. 

2. In  `database.py ` 
I also get an error in `database.py `  on this line:
`from models.book import Book`

The error says "models" can't be found. 

However, I don't get either error when I run the app without trying to query the DB.

3. Am I on the right track from using SQLAlchemy to query the DB in the view function?

In app.py:

@app.route("/books", methods=('GET'))
```
def book_list():
    """Show the list of books"""

    book_query = db_session.query(Book).filter_by(Book.id)
    return render_template("book.html", book=book_query.all())
```

Of course this code doesn't even run b/c of the errors above, but I want to check that I'm on the right track. 

4. Once the query works, I need to update the HTML to extract the information from the new query data structure. Is this correct?

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,9 +1,13 @@
 from flask import Flask, render_template, url_for
-from upcase_booklist_app.data.books import books
+
+from upcase_booklist_app.database import Base
```
Comment (paulghaddad) [inline app.py]: No need to import this, instead import the session.

File: app.py [hunk 2]
```diff
@@ -17,13 +21,20 @@ def index():
 @app.route("/books")
 def book_list():
     """Show the list of books"""
-    return render_template("books.html", books=books)
+
+    book_query = session.query(Book).filter_by(Book.id)
```
Comment (paulghaddad) [inline app.py]: Move to the end of this line: `.all()`

**Comment by amymhaddad** [comment]
@paulghaddad The changes I made are listed below. Can you please review my view functions to ensure that I'm doing the querying correctly? 

Both view functions work and return the necessary information. I want to make sure I'm using the correct data structure and the process looks correct. 

Updates in the PR:
1. In **models directory**, I updated the line in the Book class to:
from upcase_booklist_app.database import Base

2. In **database.py** I added this line to import the Book:
from upcase_booklist_app.models.book import Book

3. Updated books and book view functions in **app.py**
--->The books.html page returns all requested items in the correct style

--->The book.html page returns the correct results, but I still need to update the style. However, I first want to make sure that the view functions look good to you before I re-do the CSS.

File: app.py [hunk 3]
```diff
@@ -17,15 +23,29 @@ def index():
 @app.route("/books")
 def book_list():
     """Show the list of books"""
-    return render_template("books.html", books=books)
+
+    book_info = {}
+    for id, title, summary, url in db_session.query(Book.id, Book.title, Book.summary, Book.image_url):
```
Comment (paulghaddad) [inline app.py]: You don't even have to transform the book query result into a dictionary. You can just call `.all()` at the end and pass it into the `books` parameter on line 33. 

See how SQLAlchemy does this in the tutorial, where they call `instance.name` and `instance.fullname`. You can do the same in your template.

https://docs.sqlalchemy.org/en/14/orm/tutorial.html#querying

File: templates/book.html [hunk 4]
```diff
@@ -16,20 +16,20 @@
...
-        
-        </body>
-    </html>
\ No newline at end of file
+    <div>
+        <div class="single-book">
+            {{ book }}
+            <!-- {% for id, book_details in book.items() %}
+                {% if id == book_id %}
+                    {{ book_details['title'] }}
```
Comment (paulghaddad) [inline templates/book.html]: this is where you'd call `book.title` if you're working with a SQLAlchemy object.

**Comment by amymhaddad** [comment]
@paulghaddad I'm closing this draft pr and will open a new one.

**Pull request closed without merging**